compiler/rustc_session/src/session.rs

```diff
@@ -1199,6 +1201,18 @@ impl Session {
     ) -> Option<Symbol> {
         attrs.iter().find(|at| self.check_name(at, name)).and_then(|at| at.value_str())
     }
+
+    pub fn strip_symbols_in_osx(&self) -> Option<&str> {
```

Comment:
This seems like a strange place to put platform-specific linker args.  Is there a reason this isn't inline where it is used?

Reply:
this arguments are not specific to the linker. They are used in the path to strip symbols on macos. They didn't exist before this PR. This is where they are used:

https://github.com/rust-lang/rust/pull/82037/files#diff-862130c0af71054886913befc8084094485fa7d764d95f55644e033e34acd126R867

Am I missing anything?

Reply:
"Linker args" may have been too specific wording to use here. The idea (tied with the thought above) is that the code in `link_natively` would be something more like:

```rust
if flavor == LinkerFlavor::Darwin {
    if let Some(opt) = strip_opt(sess.opts.debugging_opts.strip) {
        let prog = Command::new("strip")
        //...
    }
}

fn strip_opt(strip: Strip) -> Option<&str> {
    match strip {
        Strip::Debuginfo => Some("-S"),
        Strip::Symbols => Some("-x"),
        Strip::None => None,
    }
}
```

That way the platform-specific stuff stays in link.rs.  I would also break things up into functions because link_natively is already a bit overwhelmingly long.

Reply:
that makes sense, thanks for the suggestion.